java/java.mx.project/test/unit/src/org/netbeans/modules/java/mx/project/SuiteCheck.java

```diff
@@ -0,0 +1,257 @@
+        log(Level.INFO, "Project found %s in %d ms", p, took);
+        assertEquals("It is suite project: " + p, "SuiteProject", p.getClass().getSimpleName());
+        OpenProjects.getDefault().open(new Project[]{p}, false);
+
+        StringBuilder errors = new StringBuilder();
+        FileObject[] errornous = { null };
+        Sources src = ProjectUtils.getSources(p);
+        int cnt = 0;
+        for (SourceGroup sourceGroup : src.getSourceGroups("java")) {
+            if (sourceGroup instanceof Compliance.Provider) {
+                Compliance c = ((Compliance.Provider) sourceGroup).getCompliance();
+                if (!c.includes(8)) {
+                    log(Level.INFO, "Skipping check of %s with compliance %s", sourceGroup, c);
+                    continue;
+                }
+            }
+            FOUND: if (onlySourceGroups.length > 0) {
+                for (String gName : onlySourceGroups) {
+                    if (sourceGroup.getDisplayName().equals(gName)) {
+                        cnt++;
+                        break FOUND;
+                    }
+                }
+                // not found
+                continue;
+            }
+            assertSourcesNoError(p, errornous, errors, sourceGroup.getRootFolder(), begin);
+        }
+        assertCompilationErrors(errors, errornous);
+
+        assertEquals("Exactly as many source groups tested as requested", onlySourceGroups.length, cnt);
+    }
+
+    protected final File findSuite(String suite) throws URISyntaxException {
+        File location = getDataDir();
+        while (location != null) {
+            File graal = new File(location, "graal");
+            File suiteDir = new File(graal, suite);
+            if (suiteDir.isDirectory()) {
+                return suiteDir;
+            }
+            location = location.getParentFile();
+        }
+        fail("Cannot find truffle next to " + getDataDir());
+        return null;
+    }
+
+    private void assertSourcesNoError(Project project, FileObject[] errornous, StringBuilder errors, FileObject dir, long begin) throws IOException {
+        long now = System.currentTimeMillis();
+        log(Level.INFO, "assertSourcesNoError for %s", dir);
+        IndexingManager.getDefault().refreshIndexAndWait(dir.toURL(), null, false);
+        log(Level.INFO, "      refresh done       %s", dir);
+        Enumeration<? extends FileObject> en = dir.getChildren(true);
+        int nonJavaCount = 0;
+        int javaCount = 0;
+        while (en.hasMoreElements()) {
+            FileObject fo = en.nextElement();
+            if (fo.isFolder()) {
+                continue;
+            }
+            Project prj = FileOwnerQuery.getOwner(fo);
+            assertSame("FileOwnerQuery returns the right project", project, prj);
+            if (!fo.hasExt("java")) {
+                nonJavaCount++;
+                continue;
+            }
+            JavaSource source = JavaSource.forFileObject(fo);
```

Comment:
Alas, this test seems to [randomly fail](https://travis-ci.org/github/apache/netbeans/jobs/749982297).

Reply:
Hopefully it fails only if the `.mx/cache` doesn't contain necessary JAR libraries (locally I have them). Let's see whether this gets fixed after 4cbe7aa